nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -466,9 +479,41 @@ protected Collection<ValidationResult> customValidate(ValidationContext validati
             );
         }
 
+        final boolean auto_commit = validationContext.getProperty(AUTO_COMMIT).asBoolean();
+        final boolean rollback_on_failure = validationContext.getProperty(RollbackOnFailure.ROLLBACK_ON_FAILURE).asBoolean();
+        if (auto_commit && rollback_on_failure) {
+            validationResults.add(new ValidationResult.Builder()
+                    .subject(RollbackOnFailure.ROLLBACK_ON_FAILURE.getDisplayName())
+                    .explanation(format("'%s' cannot be set to 'true' when '%s' is also set to 'true'. "
+                                    + "Transaction rollbacks for batch updates cannot be supported when auto commit is set to 'true'",
+                            RollbackOnFailure.ROLLBACK_ON_FAILURE.getDisplayName(), AUTO_COMMIT.getDisplayName()))
+                    .build());
+        }
+
+        if (auto_commit && !isMaxBatchSizeHardcodedToZero(validationContext)) {
```

Comment:
Variables should use camel case without the use of any underscores
```suggestion
        final boolean autoCommit = validationContext.getProperty(AUTO_COMMIT).asBoolean();
        final boolean rollbackOnFailure = validationContext.getProperty(RollbackOnFailure.ROLLBACK_ON_FAILURE).asBoolean();
        if (autoCommit && rollbackOnFailure) {
            validationResults.add(new ValidationResult.Builder()
                    .subject(RollbackOnFailure.ROLLBACK_ON_FAILURE.getDisplayName())
                    .explanation(format("'%s' cannot be set to 'true' when '%s' is also set to 'true'. "
                                    + "Transaction rollbacks for batch updates cannot be supported when auto commit is set to 'true'",
                            RollbackOnFailure.ROLLBACK_ON_FAILURE.getDisplayName(), AUTO_COMMIT.getDisplayName()))
                    .build());
        }

        if (autoCommit && !isMaxBatchSizeHardcodedToZero(validationContext)) {
```